build.gradle

```diff
@@ -169,15 +168,16 @@ allprojects {
       '-Xlint:unchecked', // Warn of type-unsafe operations on generics.
     ]
 
-    options.forkOptions.jvmArgs += jvmArgsForJavaCompilation ?: (isCI ? [] : ['-Xmx12G', '-XX:+HeapDumpOnOutOfMemoryError'])
+    options.fork = true
+    options.forkOptions.jvmArgs += jvmArgsForJavaCompilation ?: (isCI ? [] : ['-Xmx4G', '-XX:+HeapDumpOnOutOfMemoryError'])
 
     // `sourceCompatibility` and `targetCompatibility` say nothing about the Java APIs available to the compiled code.
-    // In fact, it's perfectly possible to compile Java 8 code that uses Java 11 APIs...
-    // This will work fine, until we actually try to run those compiled classes under Java 8-compatible JVM,
-    // when we'll end up with NoSuchMethodError for APIs added between Java 9 and Java 11
-    // (most notably, InputStream#readAllBytes, Stream#takeWhile and String#isBlank).
-    // `options.release = 8` makes sure that regardless of Java version used to run the compiler,
-    // only Java 8-compatible APIs are available to the compiled code.
+    // In fact, for X < Y it's perfectly possible to compile Java X code that uses Java Y APIs...
```

Comment:
Future-proofing, in case someone builds on JVM of version > 11

Reply:
have you tested that?

Reply:
Yes! for X = 8, Y = 11 - you can refer to Java 11 APIs when you build (supposedly) Java 8 code on a JVM 11

Reply:
Which is a big big bummer

Reply:
But not tested on JVM > 11... it'll take a while before JBR is upgraded from 11 to sth higher anyway

Reply:
https://github.com/VirtusLab/git-machete-intellij-plugin/issues/809